### PR TITLE
Remove 'must be non-null' and 'must not be null' comments in widgets library

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This removes all of the comments that are of the form "so-and-so (must not be null|can ?not be null|must be non-null)" from the cases where those values are defines as non-nullable values.

This PR removes them from the widgets library.

This was done by hand, since it really didn't lend itself to scripting, so it needs to be more than just spot-checked, I think. I was careful to leave any comment that referred to parameters that were nullable, but I may have missed some.

In addition to being no longer relevant after null safety has been made the default, these comments were largely fragile, in that it was easy for them to get out of date, and not be accurate anymore anyhow.

This did create a number of constructor comments which basically say "Creates a [Foo].", but I don't really know how to avoid that in a large scale change, since there's not much you can really say in a lot of cases.  I think we might consider some leniency for constructors to the "Comment must be meaningful" style guidance (which we de facto have already, since there are a bunch of these).

## Related PRs
- https://github.com/flutter/flutter/pull/134984
- https://github.com/flutter/flutter/pull/134991
- https://github.com/flutter/flutter/pull/134993
- https://github.com/flutter/flutter/pull/134994

## Tests
 - Documentation only change.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart [hunk 1]
```diff
@@ -665,8 +663,7 @@ class ListWheelScrollView extends StatefulWidget {
   /// {@macro flutter.rendering.RenderListWheelViewport.overAndUnderCenterOpacity}
   final double overAndUnderCenterOpacity;
 
-  /// Size of each child in the main axis. Must not be null and must be
-  /// positive.
+  /// Size of each child in the main axis. Must be positive.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart:666]: nit:
```suggestion
  /// Size of each child in the main axis.
  ///
  /// Must be positive.
```

File: packages/flutter/lib/src/widgets/scroll_physics.dart [hunk 2]
```diff
@@ -220,14 +220,10 @@ class ScrollPhysics {
   /// Provides a heuristic to determine if expensive frame-bound tasks should be
   /// deferred.
   ///
-  /// The velocity parameter must not be null, but may be positive, negative, or
-  /// zero.
+  /// The velocity parameter may be positive, negative, or zero.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/scroll_physics.dart:223]:
```suggestion
  /// The `velocity` parameter may be positive, negative, or zero.
```

File: packages/flutter/lib/src/widgets/scroll_physics.dart [hunk 3]
```diff
@@ -220,14 +220,10 @@ class ScrollPhysics {
...
   ///
-  /// The metrics parameter must not be null.
-  ///
-  /// The context parameter must not be null. It normally refers to the
-  /// [BuildContext] of the widget making the call, such as an [Image] widget
-  /// in a [ListView].
+  /// The context parameter normally refers to the [BuildContext] of the widget
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/scroll_physics.dart:225]:
```suggestion
  /// The `context` parameter normally refers to the [BuildContext] of the widget
```

**Review by goderbauer — APPROVED** [review]
LGTM! 🚀

File: packages/flutter/lib/src/widgets/basic.dart [hunk 4]
```diff
@@ -331,14 +328,11 @@ class Opacity extends SingleChildRenderObjectWidget {
...
-  /// An opacity of 1.0 is fully opaque. An opacity of 0.0 is fully transparent
+  /// An opacity of one is fully opaque. An opacity of zero is fully transparent
   /// (i.e., invisible).
   ///
-  /// The opacity must not be null.
-  ///
-  /// Values 1.0 and 0.0 are painted with a fast path. Other values
-  /// require painting the child into an intermediate buffer, which is
-  /// expensive.
+  /// Values one and zeri are painted with a fast path. Other values require
```
[the remark this answers is not in the document]
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/basic.dart]: It's a new number I'm promoting. It's the same as zero, but without the stigma of representing "nothing". :-)

File: packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart]: Done

File: packages/flutter/lib/src/widgets/scroll_physics.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/scroll_physics.dart]: Done

File: packages/flutter/lib/src/widgets/scroll_physics.dart (hunk 3 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/scroll_physics.dart]: Done

**Pull request merged**